pkg/kubelet/runtimeclass/runtimeclass_manager.go

```diff
@@ -35,6 +35,10 @@ type Manager struct {
 func NewManager(client clientset.Interface) *Manager {
 	const resyncPeriod = 0
 
+	if client == nil {
```

Comment:
While I agree that we should fix this, I'm not in favor of failing silently. I think we should `return nil, err` and handle the error when constructing `KubeletConfiguration`

Reply:
+1 to returning `nil, err`. 

Re what to do with that error...

One option would be to log the error in the `NewMainKubelet` function in `pkg/kubelet/kubelet.go` which calls `runtimeclass.NewManager`, and then continue processing. Alternatively, we could have `NewMainKubelet` return an error if `runtimeclass.NewManager` returns an error.

Curious for the communities opinion on whether `RuntimeClassManager` failing to start should cause the entire kubelet to fail. It seems like `NewMainKubelet` has a mix of failing/continuing with execution if individual managers fail.